src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -500,6 +500,15 @@ private void inACycleFound(@NotNull final Bytes<?> bytes) {
     }
 
     private long nextIndexWithNextAvailableCycle(final int cycle) {
+        try {
+            return nextIndexWithNextAvailableCycle0(cycle);
+        } catch (MissingStoreFileException e) {
+            queue.refreshDirectoryListing();
+            return nextIndexWithNextAvailableCycle0(cycle);
+        }
+    }
```

Comment:
This should only happen in scenarios where we used to throw an `IllegalStateException` or an `AssertionError` so it shouldn't affect the happy case, unless it impacts JIT in some mysterious way.